### PR TITLE
Fix Roomlogs crash

### PR DESCRIPTION
**Pull request by Karthik99999**

### CODE REVIEW

File: server/roomlogs.ts [hunk 1]
```diff
@@ -296,8 +296,8 @@ export class Roomlog {
 	}
 	async rename(newID: RoomID): Promise<true> {
 		await Rooms.Modlog.rename(this.roomid, newID);
-		this.roomid = newID;
 		if (roomlogTable) {
+			this.roomid = newID;
 			if (!(!Config.logchat || this.roomid.startsWith('battle-'))) {
 				await roomlogTable.updateAll({roomid: this.roomid})`WHERE roomid = ${this.roomid}`;
 			}
```
Comment (mia-pi-git) [inline server/roomlogs.ts:303]:
```suggestion
			if (!(!Config.logchat || this.roomid.startsWith('battle-'))) {
				await roomlogTable.updateAll({roomid: newID})`WHERE roomid = ${this.roomid}`;
			}
			this.roomid = newID;
```

Comment (mia-pi-git) [reply server/roomlogs.ts:303]: Also ```ts
			Roomlogs.roomlogs.delete(this.roomid);
		```
This should be somewhere before the main `if`.

File: server/roomlogs.ts [hunk 2]
```diff
@@ -313,6 +313,7 @@ export class Roomlog {
 			if (roomlogExists && !newRoomlogExists) {
 				await Monitor.logPath(roomlogPath + `/${this.roomid}`).rename(Monitor.logPath(roomlogPath + `/${newID}`).path);
 			}
+			this.roomid = newID;
```
Comment (mia-pi-git) [inline server/roomlogs.ts:316]: Strike this - it should be after the main `if` with the destroy call.

```ts
			Roomlogs.roomlogs.set(newID, this);
			this.roomid = newID;
```

File: server/roomlogs.ts (hunk 1 above)
Comment (Karthik99999) [reply server/roomlogs.ts]: If this.destroy() gets moved to the top, wouldn't that make `roomlogStreamExisted` always be false?

Comment (Karthik99999) [reply server/roomlogs.ts]: I moved that variable to before the `this.destroy()` call

**Comment by mia-pi-git** [comment]
Cool, thanks!

**Pull request merged**